### DISCUSSION TITLE
Document the exact HTTP flow when trying to access data behind earthdata login

### DISCUSSION THREAD

**Discussion by yuvipanda** [earthaccess-dev/earthaccess#851] [Ideas]
We made an attempt at this in https://hackmd.io/T73AtFTnS4C_Ez9JfGNldA.

The goal is to try to describe *exactly* the HTTP requests and responses that happen whenever we try to access earthdata login protected data. This can and does differ based on the following criteria:

1. Location of the client making the request (AWS us-west-2 or not seems to be the primary differentiator)
2. Wether the data is on AWS or out of AWS
3. Particular DAACs hosting the data and choices they have made
4. The application responsible for doing the OAuth2 flow between the source of the data and earthdata - this could be TEA, Cumulus, or anything else really.

Doing this at the HTTP protocol layer really helps us understand where problems are and how we can solve them - not just in python, but in all other languages.

**Comment by JessicaS11** [comment]
Requests, auth, etc. is all a bit of a black box to me, but I'm trying to use earthaccess to get the proper credentials to then order/download ICESat-2 data (for within icepyx). I can log in with earthaccess, but try as I might, I cannot figure out how to pass that authentication to NSIDC's EGI (the ICESat-2 subsetter, [base url](https://n5eil02u.ecs.nsidc.org/egi/request)). So far, everything I've tried results in a 401 authentication error (or simply no content in the response). I have no idea if I'm somehow not passing credentials correctly or if it really is somehow not accepting my credentials. Happy to try and note this here or elsewhere as a use case, and would appreciate any feedback on what to try.
